lambdas/src/handlers/database-user-initializer/update.ts

```diff
@@ -0,0 +1,58 @@
+import * as _ from 'lodash';
+import {CloudFormationCustomResourceUpdateEvent} from 'aws-lambda';
+import {Database} from '../database';
+import {SecretsManager} from '../secrets-manager';
+import {CustomResourceHandler} from '../base';
+import {ScriptBuilder} from '../script-builder';
+
+export class DatabaseUserInitializerUpdateHandler extends CustomResourceHandler<CloudFormationCustomResourceUpdateEvent> {
+    public async consumeEvent() {
+        const props = this.event.ResourceProperties;
+        const oldProps = this.event.OldResourceProperties;
+
+        const credential = await SecretsManager.getDatabaseCredential({
+            region: props.Region,
+            secretId: props.DatabaseAdminUserSecretName
+        });
+
+        let script = '';
+
+        if (props.DatabaseAdminUserSecretName === oldProps.DatabaseAdminUserSecretName) {
+            const removedUsers: any = _.differenceBy(oldProps.DatabaseUsers, props.DatabaseUsers, 'username');
+            for (const removedUser of removedUsers) {
+                script = script.concat(await ScriptBuilder.dropUserScript(removedUser));
+            }
+
+            const addedUsers: any = _.differenceBy(props.DatabaseUsers, oldProps.DatabaseUsers, 'username');
+            for (const addedUser of addedUsers) {
+                script = script.concat(await ScriptBuilder.createUserScript(addedUser, credential.dbname, props.Region));
+            }
+
+            for (const newUser of props.DatabaseUsers) {
+                for (const oldUser of oldProps.DatabaseUsers) {
```

Comment:
Is it possible to do this without having to do a nested loop? Could it be possible to handle this by using `indexOf` over the object's `username` field?